libs/pageserver_api/src/shard.rs

```diff
@@ -438,6 +460,57 @@ impl<'de> Deserialize<'de> for ShardIndex {
     }
 }
 
+/// Whether this key should be ingested by all shards
+fn key_is_broadcast(key: &Key) -> bool {
+    // TODO: can we be less conservative?  Starting point is to broadcast everything
+    // except for rel block keys
+    !is_rel_block_key(key)
```

Comment:
This function doesn't do much in this PR: it starts to matter when we actually split the WAL (next PR https://github.com/neondatabase/neon/pull/6024).

Yes, it will need updating+renaming to reflect that we're not really broadcasting these "not sharded" keys, but rather just sending them to shard 0.

Reply:
Cleaned this up to clearly refer to shard 0 as the special case, rather than broadcast, in f30739e491e104be170564a7945286c8539804ed